### PR TITLE
v1.2.0-rc1 WIP

### PR DESCRIPTION
**Pull request by AnwarShahriar**

### CODE REVIEW

**Comment by AnwarShahriar** [comment]
@bfranks all the issues are acknowledged in this PR.

File: waveformview/src/main/java/space/siy/waveformview/FixedWaveFormView.kt [hunk 1]
```diff
@@ -48,34 +50,113 @@ class FixedWaveFormView(
...
+    blockPaint = Paint()
+    domeRadius = blockWidth / 2
+    lp.recycle()
+  }
+
+  private var upperWaveBars: Array<RectF>? = null
+  private var bottomWaveBars: Array<RectF>? = null
+
+  // api
+  var duration: Long = 1L // We use this in some division, so don't make it zero
```
Comment (bfranks) [inline waveformview/src/main/java/space/siy/waveformview/FixedWaveFormView.kt]: why 1L and not -1L since -1 should be an invalid value?

Comment (AnwarShahriar) [reply waveformview/src/main/java/space/siy/waveformview/FixedWaveFormView.kt]: This is used in some division, which updates the ui. Resulting in negative value can create a bunch of ui glitches and break some assumptions in those part.

File: waveformview/src/main/java/space/siy/waveformview/FixedWaveFormView.kt [hunk 2]
```diff
@@ -48,34 +50,113 @@ class FixedWaveFormView(
...
 
   /**
    * WaveFormData show in view
    */
-  var data: WaveFormData? = null
+  var waveFormData: WaveFormData? = null
     set(value) {
       field = value
+      duration = value?.duration ?: 1L
```
Comment (bfranks) [inline waveformview/src/main/java/space/siy/waveformview/FixedWaveFormView.kt]: Make this respect above comment about 0.

**Review by bfranks — APPROVED** [review]

**Pull request merged**